src/Kiota.Builder/Writers/TypeScript/CodeConstantWriter.cs

```diff
@@ -11,16 +11,37 @@ public override void WriteCodeElement(CodeConstant codeElement, LanguageWriter w
     {
         ArgumentNullException.ThrowIfNull(codeElement);
         ArgumentNullException.ThrowIfNull(writer);
-        if (codeElement.OriginalInterface is null) throw new InvalidOperationException("Original interface cannot be null");
-        writer.StartBlock($"const {codeElement.Name.ToFirstCharacterLowerCase()}: Record<string, string> = {{");
-        foreach (var property in codeElement.OriginalInterface
-                                                .Properties
-                                                .OfKind(CodePropertyKind.QueryParameter)
-                                                .Where(static x => !string.IsNullOrEmpty(x.SerializationName))
-                                                .OrderBy(static x => x.SerializationName, StringComparer.OrdinalIgnoreCase))
+        if (codeElement.OriginalCodeElement is null) throw new InvalidOperationException("Original CodeElement cannot be null");
+        switch (codeElement.Kind)
         {
-            writer.WriteLine($"\"{property.Name.ToFirstCharacterLowerCase()}\": \"{property.SerializationName}\",");
+            case CodeConstantKind.QueryParametersMapper:
+                if (codeElement.OriginalCodeElement is not CodeInterface codeInterface) throw new InvalidOperationException("Original CodeElement cannot be null");
```

Comment:
can you move the bodies of the two cases to dedicated methods to ease up maintenance later please?